### PR TITLE
feat(FEC-11789): Update to the latest IMA SDK OMID API

### PR DESCRIPTION
**Pull request by JonathanTGold**
### Description of the Changes

Update to the latest IMA SDK OMID API

solves: FEC-11789

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: flow-typed/types/ima-config.js [hunk 1]
```diff
@@ -56,7 +55,7 @@ type _ImaConfigObject = {
   adBreakCuePointStyle: Object,
   sessionId: string,
   enableOmidBeta: boolean,
```
Comment (yairans) [inline flow-typed/types/ima-config.js:57]: remove enableOmidBeta

File: flow-typed/types/omid-access-modes-config.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * @type { 'FULL' | 'CREATIVE' | 'DOMAIN' | 'LIMITED'}
+ */
+declare type AccessMode = 'FULL' | 'CREATIVE' | 'DOMAIN' | 'LIMITED';
+
+/**
+ * {@link https://developers.google.com/interactive-media-ads/docs/sdks/html5/client-side/reference/js/google.ima#.OmidVerificationVendor different verification script providers}
+ * @typedef OmidVendor
+ * @type { 'OTHER' | 'MOAT' | 'DOUBLEVERIFY' | 'INTEGRAL_AD_SCIENCE'  | 'PIXELATE'  | 'NIELSEN'  | 'COMSCORE' | 'MEETRICS' | 'GOOGLE'}
+ */
+declare type OmidVendor = 'OTHER' | 'MOAT' | 'DOUBLEVERIFY' | 'INTEGRAL_AD_SCIENCE'  | 'PIXELATE'  | 'NIELSEN'  | 'COMSCORE' | 'MEETRICS' | 'GOOGLE';
```
Comment (yairans) [inline flow-typed/types/omid-access-modes-config.js:15]: I would keep it as string and just refer to the list in ima docs. so we won't need to change it on ima updates.
same for AccessMode

File: src/ima.js [hunk 3]
```diff
@@ -886,10 +883,10 @@ class Ima extends BasePlugin implements IMiddlewareProvider, IAdsControllerProvi
...
         adsRequest.vastLoadTimeout = this.config.vastLoadTimeout;
       }
-      if (typeof this.config.omSdkAccessModes === 'object') {
-        Object.keys(this.config.omSdkAccessModes).forEach(accessMode => {
-          adsRequest.omidAccessModeRules[accessMode] = this.config.omSdkAccessModes[accessMode];
-        });
+      if (typeof this.config.omidAccessModes === 'object') {
+        for (const [vendor: OmidVendor, accessMode: AccessMode] of Object.entries(this.config.omidAccessModes)) {
+          adsRequest.omidAccessModeRules[this._sdk.OmidVerificationVendor[vendor]] = this._sdk.OmidAccessMode[accessMode];
+        }
```
Comment (yairans) [inline src/ima.js:889]:
```suggestion
adsRequest.omidAccessModeRules = {...adsRequest.omidAccessModeRules, ...this.config.omidAccessModes}
```

**Review by yairans — CHANGES_REQUESTED** [review]

File: src/ima.js (hunk 3 above)
Comment (JonathanTGold) [reply src/ima.js:889]: you will end with the kies instead of the values 
"this._sdk.OmidVerificationVendor[vendor] " return numerous values

**Review by yairans — APPROVED** [review]

**Pull request merged**